partner_company_group/models/res_partner.py

```diff
@@ -18,3 +18,64 @@ class Contact(models.Model):
 
     def _commercial_fields(self):
         return super()._commercial_fields() + ["company_group_id"]
+
+    @api.onchange("company_group_id")
+    def _onchange_company_group_id(self):
+        res = {}
+        if (
+            self.company_group_id
+            and self.company_group_id.property_product_pricelist
+            != self.property_product_pricelist
+        ):
+            res["warning"] = {
+                "title": _("Warning"),
+                "message": _(
+                    "The company group %s has the pricelist %s, that is different than"
+                    " the pricelist set on this contact"
+                    % (
+                        self.company_group_id.display_name,
+                        self.company_group_id.property_product_pricelist.display_name,
+                    )
+                ),
+            }
+        return res
+
+    @api.onchange("property_product_pricelist")
+    def _onchange_property_product_pricelist(self):
+        res = {}
+        if (
+            self.company_group_id
+            and self.company_group_id.property_product_pricelist
+            != self.property_product_pricelist
+        ):
+            res["warning"] = {
+                "title": _("Warning"),
+                "message": _(
+                    "The company group %s has the pricelist %s, that is different than"
+                    " the pricelist set on this contact"
+                    % (
+                        self.company_group_id.display_name,
+                        self.company_group_id.property_product_pricelist.display_name,
+                    )
+                ),
+            }
+        elif (
```

Comment:
You could extract the common code to a new method to not repeat the same code ([DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)).
Or in this case call direcly to _onchange_company_group_id
```suggestion
        res = self._onchange_company_group_id()
        if not res:
```